DashAISoftware/DashAI
# Edit dataset

# Summary


Includes a modal to change the name and task associated with a dataset

## Type of change



- Front end new feature.

## Changes


- Add a modal with a TextField to change the name of a dataset and also renders `ItemSelector.jsx` to change the task associated to the dataset
- updates the Datasets table once the update is done

## How to Test


Option 1 - Backend
1. Go to `DashAI/DashAI/front` and run `yarn build`
2. Go back to `DashAI/` and run `python -c "import DashAI;DashAI.run()`
3. Navigate to the datasets tab
4.  Click the button to edit dataset (requires at least one dataset in the table)
5. Modify the name and the task and click "Save"
6. The modified values should be updated in the datasets table.

Option 2 - json-server
1. Go to `DashAI/DashAI/front` and run `yarn json-server`
2. in another terminal in the same folder of step 1 run `yarn start`
4. Navigate to the datasets tab
5.  Click the button to edit dataset (requires at least one dataset in the table)
6. Modify the name and the task (this option doesn't support updating the values in the table)

## Screenshots


![edit-dataset](https://github.com/DashAISoftware/DashAI/assets/53657198/83587ec6-be63-4fe9-ba3f-933846f9f155)

## Notes


This PR depends on #75
